pimd/pim6_mld.c

```diff
@@ -2231,13 +2231,558 @@ void gm_ifp_update(struct interface *ifp)
+			&p->query, &p->expiry);
+	}
+	vty_out(vty, "  %zu expiry timers from *,G queries\n",
+		gm_grp_pends_count(gm_ifp->grp_pends));
+	vty_out(vty, "  %zu expiry timers from S,G queries\n",
+		gm_gsq_pends_count(gm_ifp->gsq_pends));
+	vty_out(vty, "  %zu total *,G/S,G from %zu hosts in %zu bundles\n",
+		gm_sgs_count(gm_ifp->sgs),
+		gm_subscribers_count(gm_ifp->subscribers),
+		gm_packet_expires_count(gm_ifp->expires));
+	vty_out(vty, "\n");
+}
+
+static void gm_show_if_one(struct vty *vty, struct interface *ifp,
+			   json_object *js_if)
+{
+	struct pim_interface *pim_ifp = (struct pim_interface *)ifp->info;
+	struct gm_if *gm_ifp = pim_ifp->mld;
+	bool querier;
+
+	if (!gm_ifp) {
+		if (js_if)
+			json_object_string_add(js_if, "state", "down");
+		else
+			vty_out(vty, "%-16s  %5s\n", ifp->name, "down");
+		return;
+	}
+
+	querier = IPV6_ADDR_SAME(&gm_ifp->querier, &pim_ifp->ll_lowest);
+
+	if (js_if) {
+		json_object_string_add(js_if, "state", "up");
+		json_object_string_addf(js_if, "version", "%d",
+					gm_ifp->cur_version);
+		json_object_string_addf(js_if, "upTime", "%pTVMs",
+					&gm_ifp->started);
+		json_object_boolean_add(js_if, "querier", querier);
+		json_object_string_addf(js_if, "querierIp", "%pPA",
+					&gm_ifp->querier);
+		if (querier)
+			json_object_string_addf(js_if, "queryTimer", "%pTH",
+						gm_ifp->t_query);
+		else
+			json_object_string_addf(js_if, "otherQuerierTimer",
+						"%pTH",
+						gm_ifp->t_other_querier);
+	} else {
+		vty_out(vty, "%-16s  %-5s  %d  %-25pPA  %-5s %11pTH  %pTVMs\n",
+			ifp->name, "up", gm_ifp->cur_version, &gm_ifp->querier,
+			querier ? "query" : "other",
+			querier ? gm_ifp->t_query : gm_ifp->t_other_querier,
+			&gm_ifp->started);
+	}
+}
+
+static void gm_show_if_vrf(struct vty *vty, struct vrf *vrf, const char *ifname,
+			   bool detail, json_object *js)
+{
+	struct interface *ifp;
+	json_object *js_vrf;
+
+	if (js) {
+		js_vrf = json_object_new_object();
+		json_object_object_add(js, vrf->name, js_vrf);
+	}
+
+	FOR_ALL_INTERFACES (vrf, ifp) {
+		json_object *js_if = NULL;
+
+		if (ifname && strcmp(ifp->name, ifname))
+			continue;
+		if (detail && !js) {
+			gm_show_if_one_detail(vty, ifp);
+			continue;
+		}
+
+		if (!ifp->info)
+			continue;
+		if (js) {
+			js_if = json_object_new_object();
+			json_object_object_add(js_vrf, ifp->name, js_if);
+		}
+
+		gm_show_if_one(vty, ifp, js_if);
+	}
+}
+
+static void gm_show_if(struct vty *vty, struct vrf *vrf, const char *ifname,
+		       bool detail, json_object *js)
+{
+	if (!js && !detail)
+		vty_out(vty, "%-16s  %-5s  V  %-25s  %-18s  %s\n", "Interface",
+			"State", "Querier", "Timer", "Uptime");
+
+	if (vrf)
+		gm_show_if_vrf(vty, vrf, ifname, detail, js);
+	else
+		RB_FOREACH (vrf, vrf_name_head, &vrfs_by_name)
+			gm_show_if_vrf(vty, vrf, ifname, detail, js);
+}
+
+DEFPY(gm_show_interface,
+      gm_show_interface_cmd,
+      "show ipv6 mld [vrf <VRF|all>$vrf_str] interface [IFNAME] [detail$detail|json$json]",
```

Comment:
Document needs to be updated for the new CLIs added here. It can be a separate PR but should be added.